packages/amplify_datastore/example/integration_test/model_type_test.dart

```diff
@@ -0,0 +1,426 @@
+        models: models,
+        classType: TimeTypeModel.classType,
+        // should be unskipped after https://github.com/aws-amplify/amplify-flutter/issues/817 is resolved
+        skips: {DataStoreOperation.query: true},
+      );
+    });
+
+    group('TemporalTime (with null value)', () {
+      var models = List.generate(5, (i) => TimeTypeModel());
+      testModelOperations<TimeTypeModel>(
+        models: models,
+        classType: TimeTypeModel.classType,
+      );
+    });
+
+    group('List<TemporalTime>', () {
+      var now = DateTime.now();
+      var list =
+          List.generate(3, (i) => TemporalTime(now.add(Duration(days: i))));
+      var models = List.generate(5, (_) => TimeListTypeModel(value: list));
+      testModelOperations<TimeListTypeModel>(
+        models: models,
+        classType: TimeListTypeModel.classType,
+      );
+    });
+
+    group(
+      'List<TemporalTime> (with null value)',
+      () {
+        var models = List.generate(5, (_) => TimeListTypeModel());
+        testModelOperations<TimeListTypeModel>(
+          models: models,
+          classType: TimeListTypeModel.classType,
+        );
+      },
+      // unskip when https://github.com/aws-amplify/amplify-flutter/issues/813 is resolved
+      skip: true,
+    );
+
+    group('TemporalTimestamp', () {
+      var now = DateTime.now();
+      var models = List.generate(
+          5, (i) => TimestampTypeModel(value: TemporalTimestamp(now)));
+      testModelOperations<TimestampTypeModel>(
+        models: models,
+        classType: TimestampTypeModel.classType,
+      );
+    });
+
+    group('TemporalTimestamp (with null value)', () {
+      var models = List.generate(5, (i) => TimestampTypeModel());
+      testModelOperations<TimestampTypeModel>(
+        models: models,
+        classType: TimestampTypeModel.classType,
+      );
+    });
+
+    group(
+      'List<TemporalTimestamp>',
+      () {
+        var now = DateTime.now();
+        var list = List.generate(
+            3, (i) => TemporalTimestamp(now.add(Duration(days: i))));
+        var models =
+            List.generate(5, (_) => TimestampListTypeModel(value: list));
+        testModelOperations<TimestampListTypeModel>(
+          models: models,
+          classType: TimestampListTypeModel.classType,
+        );
+      },
+      // should be unskipped after https://github.com/aws-amplify/amplify-flutter/issues/814 is resolved
+      skip: true,
+    );
+
+    group(
+      'List<TemporalTimestamp> (with null value)',
+      () {
+        var models = List.generate(5, (_) => TimestampListTypeModel());
+        testModelOperations<TimestampListTypeModel>(
+          models: models,
+          classType: TimestampListTypeModel.classType,
+        );
+      },
+      // unskip when https://github.com/aws-amplify/amplify-flutter/issues/813 is resolved
+      skip: true,
+    );
+
+    group('AWSJSON', () {
+      String json = jsonEncode(Map.from({
```

Comment:
Don't need `Map.from` here - just creates another map from the map arg. Main use case is retyping a map, i.e.

```dart
Map map1 = {'a': 'b', 'c': 'd'}; // Map<dynamic, dynamic>
Map<String, String> map2 = Map.from(map1); // creates copy + asserts k-v types
```